### ISSUE TITLE
auto-purge-failures & purge-policy

### ISSUE THREAD

**Issue by soulen3** [holland-backup/holland#269]
If you have the following configuration options a failed backup can still be purged
auto-purge-failures = yes
purge-policy = before-backup

**Comment by soulen3** [comment]
After debugging this, I found that the current backup directory was being purged.  This caused Holland to report an error connecting to the database. (It was missing it's cnf file, and use default settings)

The root cause of all this was a `after-backup-command` that renamed the current backup directory to 'nightly'.  The purge is expecting the directories in /var/spool/holland/<backupset> to name based on the date, so sorting them always put nightly at the bottom of the list.  So with `backups-to-keep=0` and 'nightly' sorted after it, the current backup was purged.

```
[holland:backup]
plugin = xtrabackup
backups-to-keep = 0
auto-purge-failures = yes
purge-policy = before-backup
```

**Comment by soulen3** [comment]
Fixed in 1.1.12
